src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -0,0 +1,170 @@
+  );
+
+  const scale = scaleLinear()
+    .domain([1, length])
+    .range([1, props.width])
+    .clamp(true);
+
+  return (
+    <div className={styles.container}>
+      {/* TODO: The sorting needs to be done based on the `score`? once it is available */}
+      {Object.keys(proteinDomainsResources)
+        .sort()
+        .map((type, key) => {
+          return (
+            <div key={key} className={styles.resourceGroup}>
+              <div className={styles.resourceName}>{type}</div>
+              <div className={styles.resourceImages}>
+                {Object.keys(proteinDomainsResources[type])
+                  .sort()
+                  .map((description, key) => {
+                    return (
+                      <div key={key} className={styles.resourceImage}>
+                        <svg
+                          className={styles.containerSvg}
+                          width={props.width}
+                          height={BLOCK_HEIGHT}
+                        >
+                          <g>
+                            <Track {...props} scale={scale} />
+                            {proteinDomainsResources[type][description].map(
+                              (domain, index) => (
+                                <DomainBlock
+                                  key={index}
+                                  domain={domain}
+                                  transcriptStart={transcriptStart}
+                                  className={props.classNames?.domain}
+                                  scale={scale}
+                                />
+                              )
+                            )}
+                          </g>
+                        </svg>
+                        <div className={styles.resourceDescription}>
+                          {description}
+                        </div>
+                      </div>
+                    );
+                  })}
+              </div>
+            </div>
+          );
+        })}
+    </div>
+  );
+};
+
+const Track = (
+  props: ProteinDomainImageProps & { scale: ScaleLinear<number, number> }
+) => {
+  const trackClasses = classNames(styles.track, props.classNames?.track);
+
+  // we are adjusting the width of every track segment by adding 2 points from its width.
+  return (
+    <g className={trackClasses}>
+      <rect height={TRACK_HEIGHT} width={props.width + 2} />
+    </g>
+  );
+};
+
+type DomainBlockProps = {
+  domain: {
+    start: number;
+    end: number;
+  };
+
+  transcriptStart: number;
+  className?: string;
+  scale: ScaleLinear<number, number>;
+};
+
+const DomainBlock = (props: DomainBlockProps) => {
+  const y = 3;
+  const domainClasses = classNames(styles.domain, props.className);
+
+  return (
+    <rect
+      key={props.domain.start}
+      className={domainClasses}
+      y={y}
+      height={BLOCK_HEIGHT}
+      x={props.scale(props.domain.start)}
+      width={props.scale(props.domain.end - props.domain.start)}
```

Comment:
`props.scale(props.domain.end - props.domain.start + 1)`?

not that we'll see a change in the visualisation, but still?

Reply:
I remember putting + 1 here but lost it somehow. Will add it back.

Reply:
That's done